.github/workflows/generate-changelog.yml

```diff
@@ -0,0 +1,30 @@
+name: Generate CHANGELOG
+on:
+  pull_request:
+    types: [closed]
+  workflow_dispatch:
+jobs:
+  GenerateChangelog:
+    if: github.event.pull_request.merged || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+        with:
+          go-version-file: 'go.mod'
+      - run: make tools update-changelog-unreleased-section
+      - run: |
+          if [[ $(git status --porcelain) ]]; then
+            if ${{github.event_name == 'workflow_dispatch'}}; then
+              MSG="Update CHANGELOG.md (Manual Trigger)"
+            else
+              MSG="Update CHANGELOG.md for #${{ github.event.pull_request.number }}"
+            fi
+            git config --local user.email changelogbot@hashicorp.com
+            git config --local user.name changelogbot
+            git add CHANGELOG.md
+            git commit -m "$MSG"
+            git push
```

Comment:
what happens if there are not entries yet? (e.g. in a new release and the PR is a chore with no entry file)

is an empty commit pushed?

Reply:
Nothing would happen. 
`update-changelog-unreleased-section.sh` only updates changelog if there are new entries. If the script does not do any changes to changelog the script would not enter inside the outermost if where the commit and push are performed`if [[ $(git status --porcelain) ]]; then`